docs/proposals/20210310-opt-in-autoscaling-from-zero.md

```diff
@@ -175,6 +179,7 @@ const (
 // DockerMachineTemplateStatus defines the observed state of a DockerMachineTemplate
 type DockerMachineTemplateStatus struct {
     Capacity corev1.ResourceList `json:"capacity,omitempty"`
+    NodeInfo corev1.NodeSystemInfo `json:"nodeInfo,omitempty"`
```

Comment:
Does embedding `corev1.NodeSystemInfo` make sense? We may only need a subset of it, xref:

https://github.com/kubernetes/api/blob/master/core/v1/types.go#L6165-L6190

The fields in `corev1.NodeSystemInfo` don't specify omitEmpty, so it would get very verbous already when only requiring e.g. a single value.

Reply:
Also: at least should be a pointer when using the struct directly.

Reply:
@chrischdi, you're right, thanks. I'll change it to a pointer. 

As for using a different struct, I am open to both the solutions. The advantage of using the NodeSystemInfo struct is to bind it with the pod status field using the same key name and type. On the other side, yes, we can have a dedicated struct and only keeps the subset of fields that the cluster API project needs.

Reply:
i think it's important to note that this information will get transformed into the node template for the core autoscaler. i'm not sure how, or which values, are used by the scheduler. that might be worth investigating.

Reply:
I'm not 100% sure, but AFAIK, the scheduler doesn't directly use the information stored in $node.status.nodeInfo.

I understand the concern about the additional fields to be stored in the KV store, and I'm cool with either solution. 

If the community agrees to a dedicated structure, I can update the proposal. 

In that case, and based on other controllers I've seen around: 
- Would it be okay to have a package `platform` in cluster-api/util, or do you have better ideas of where this struct should belong? 
- Do we want to export it for the status reconcilers, or do we want to delegate the providers' developers to develop their own? 
 
The cluster autoscaler, on the other side, should consume this information from the object deserialized as unstructured.

Reply:
Another alternative is to use a dedicated struct to cover the labels/taint capacity annotation from the status field too.

Reply:
Based on our discussion yesterday during the meeting, I've updated the proposal to use an ad-hoc struct to implement in this repository's `util` package that providers can import to build the status field `nodeInfo`.

@elmiko @chrischdi @JoelSpeed PTAL

Thanks all for your availability to discuss this during the meeting yesterday :)

Reply:
I left a few comments below. I think this goes in the right direction